### PR TITLE
Push marks

### PR DESCRIPTION
**Pull request by DamienCassou**

### CODE REVIEW

**Comment by DamienCassou** [comment]
Any feedback on this one @Fuco1?

File: beginend.el [hunk 1]
```diff
@@ -49,25 +49,32 @@
...
+       (goto-char ,extremum-var)
+       ,@body
+       (if (equal ,oldpos-var (point))
+           (goto-char ,extremum-var)
+         (unless (eq ,oldpos-var ,extremum-var)
+           (push-mark ,oldpos-var))))))
+
 (defmacro beginend--double-tap-begin (&rest body)
-  "Evaluate &BODY and go-to real beginning if that did not change point."
+  "Evaluate BODY and go-to `point-min' if that did not change point."
```
Comment (Fuco1) [inline beginend.el]: It is a bit unclear what *that* is.  I guess `if point did not change/move` might be slightly more clear?

File: beginend.el [hunk 2]
```diff
@@ -49,25 +49,32 @@
...
+  "Go to point EXTREMUM if executing BODY did not change point."
+  (declare (debug (form body))
+           (indent 1))
+  (let ((oldpos-var (make-symbol "old-position"))
+        (extremum-var (make-symbol "extremum")))
+    `(let ((,oldpos-var (point))
+           (,extremum-var ,extremum))
+       (goto-char ,extremum-var)
+       ,@body
+       (if (equal ,oldpos-var (point))
```
Comment (Fuco1) [inline beginend.el]: Use `=` for numbers

File: beginend.el [hunk 3]
```diff
@@ -49,25 +49,32 @@
...
+           (goto-char ,extremum-var)
+         (unless (eq ,oldpos-var ,extremum-var)
```
Comment (Fuco1) [inline beginend.el]: Use `=` for numbers.  Maybe change to `when /=`, I'm not sure which reads better, might be too subjective.

**Comment by Fuco1** [comment]
Pretty clever to only push the mark from the original jump and not on the begin/realbegin back-and-forth.  I haven't thought of that :)

**Comment by DamienCassou** [comment]
I took your feedback into account. Please merge if you like it.

**Pull request merged**